// нет, catch обработает ошибку в f1
